crypto/crypto.go

```diff
@@ -258,28 +240,28 @@ func Encrypt(ks *Key, ciphertext, plaintext []byte) ([]byte, error) {
 // Decrypt verifies and decrypts the ciphertext. Ciphertext must be in the form
 // IV || Ciphertext || MAC. plaintext and ciphertext may point to (exactly) the
 // same slice.
-func Decrypt(ks *Key, plaintext, ciphertext []byte) ([]byte, error) {
+func Decrypt(ks *Key, plaintext []byte, ciphertextWithMac []byte) ([]byte, error) {
```

Comment:
I found it super confusing that `ciphertext` is being overwritten several times (first with mac and iv, then just with mac, then just ciphertext). So unless that has some performance reasons, I think it's better to be explicit here, since the crypto code is easier to audit the more readable it is.